src/AdoPat/PatCache.cs

```diff
@@ -0,0 +1,81 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License.
+
+namespace Microsoft.Authentication.AdoPat
+{
+    using System;
+    using System.Collections.Generic;
+    using System.Text;
+    using System.Text.Json;
+    using Microsoft.VisualStudio.Services.DelegatedAuthorization;
+
+    /// <summary>
+    /// A simple cache mapping <see cref="string"/> to <see cref="PatToken"/> backed by secure storage.
+    /// </summary>
+    public class PatCache
+    {
+        private IStorageWrapper storage;
+        private Lazy<Dictionary<string, PatToken>> cache;

```

Comment:
What's the purpose of using Lazy Dictionary here?

Reply:
Using `Lazy<T>` here is a clean way to provide memoization of the cache itself, and has the benefit of moving the computation to read it to first use, instead of when creating the PatCache instance.

Reply:
@kyle-rader is exactly right. I just used `Lazy<T>` as a way of avoiding a read of the storage in the constructor for this class as I generally don't like to see I/O in constructors.